### PR TITLE
perf(skymp5-server): optimize memory allocation in pex reader

### PR DESCRIPTION
**Pull request by bogdasar1985**

### CODE REVIEW

File: skymp5-server/cpp/papyrus_vm_lib/Reader.cpp [hunk 1]
```diff
@@ -13,7 +14,9 @@ void Reader::Read()
   if (File.is_open()) {
 
     File.seekg(0, std::ios_base::end);
+    fl_size = File.tellg();
```
Comment (nic11) [inline skymp5-server/cpp/papyrus_vm_lib/Reader.cpp:17]: I think there's no reason to define it in the beginning, could say `const auto fileSize = ...` right here.

File: skymp5-server/cpp/papyrus_vm_lib/Reader.cpp [hunk 2]
```diff
@@ -13,7 +14,9 @@ void Reader::Read()
...
     File.seekg(0, std::ios_base::beg);
+    arrayBytes.reserve(fl_size);  // Желательно, чтобы ближайшая степень двойки была, но с лимитом в 4096
```
Comment (nic11) [inline skymp5-server/cpp/papyrus_vm_lib/Reader.cpp:19]: 1. Sounds more like a self-note, should not go to final commit.
2. No reason to round up to be power of two and/or divisible by 4096, allocators should do that by themselves. If only we care a lot about caching, but no reason to overoptimize stuff here, it requires a more global approach targeting entire project to have any effect.

P.S. Let's also keep all comments, including temporrary ones, in English

**Review by nic11 — APPROVED** [review]

File: skymp5-server/cpp/papyrus_vm_lib/Reader.cpp [hunk 3]
```diff
@@ -118,6 +121,7 @@ StringTable Reader::FillStringTable()
   std::vector<std::string> storage;
 
   int SizeStringTable = Read16_bit();
+  storage.reserve(SizeStringTable);
```
Comment (Pospelove) [inline skymp5-server/cpp/papyrus_vm_lib/Reader.cpp:124]: This one also should be fixed even if this variable wasn't created by this PR. I think the more convenient name would be `stringTableSize`

File: skymp5-server/cpp/papyrus_vm_lib/Reader.cpp [hunk 4]
```diff
@@ -477,6 +481,7 @@ uint64_t Reader::Read64_bit()
 std::string Reader::ReadString(int Size)
 {
   std::string temp;
+  temp.reserve(Size);
```
Comment (Pospelove) [inline skymp5-server/cpp/papyrus_vm_lib/Reader.cpp:484]: Please replace `Size` with `size`

File: skymp5-server/cpp/papyrus_vm_lib/Reader.cpp [hunk 5]
```diff
@@ -13,7 +14,9 @@ void Reader::Read()
   if (File.is_open()) {
 
     File.seekg(0, std::ios_base::end);
+    fileSize = File.tellg();
```
Comment (Pospelove) [inline skymp5-server/cpp/papyrus_vm_lib/Reader.cpp:17]: Please replace `File` with `file`

File: skymp5-server/cpp/papyrus_vm_lib/Reader.cpp [hunk 6]
```diff
@@ -5,6 +5,7 @@ void Reader::Read()
 {
 
   char temp;
+  std::streampos fileSize;
```
Comment (Pospelove) [inline skymp5-server/cpp/papyrus_vm_lib/Reader.cpp:8]: Consider moving these definitions below to the place they are actually used

**Review by Pospelove — CHANGES_REQUESTED** [review]

**Review by Pospelove — APPROVED** [review]

**Pull request merged**